modules/tinymce/src/core/main/ts/api/WindowManager.ts

```diff
@@ -78,20 +82,27 @@ const WindowManager = (editor: Editor): WindowManager => {
     });
   };
 
-  const addDialog = <T extends Dialog.DialogData>(dialog: InstanceApi<T>) => {
-    dialogs.push(dialog);
+  const addDialog = <T extends Dialog.DialogData>(dialog: InstanceApi<T>, triggerElement: Optional<SugarElement<HTMLElement>>) => {
+    dialogs.push({ instanceApi: dialog, triggerElement });
     fireOpenEvent(dialog);
   };
 
   const closeDialog = <T extends Dialog.DialogData>(dialog: InstanceApi<T>) => {
     fireCloseEvent(dialog);
-    dialogs = Arr.filter(dialogs, (otherDialog) => {
-      return otherDialog !== dialog;
+
+    Arr.findIndex(dialogs, ({ instanceApi }) => instanceApi === dialog).each( (dialogIndex) => {
+      const dialogTriggerElement = dialogs[dialogIndex].triggerElement;
+
+      dialogs.splice(dialogIndex, 1);
+
+      // Move focus back to editor when the last window is closed
+      if (dialogs.length === 0) {
+        editor.focus();
+      } else {
+        // Move focus to the element that was active before the dialog was opened
+        dialogTriggerElement.each((el) => Focus.focus(el));
```

Comment:
Retaining references to DOM elements can be a tricky thing and sometimes lead to memory leaks. Just to be safe, check the element is still attached before trying to focus it.
```suggestion
        dialogTriggerElement.filter(SugarBody.inBody).each(Focus.focus);
```